components/config/opts.rs

```diff
@@ -1104,15 +1091,26 @@ pub fn get() -> RwLockReadGuard<'static, Opts> {
 pub fn parse_pref_from_command_line(pref: &str) {
     let split: Vec<&str> = pref.splitn(2, '=').collect();
     let pref_name = split[0];
-    let value = split.get(1);
-    match value {
-        Some(&"false") => PREFS.set(pref_name, PrefValue::Boolean(false)),
-        Some(&"true") | None => PREFS.set(pref_name, PrefValue::Boolean(true)),
-        Some(value) => match value.parse::<f64>() {
-            Ok(v) => PREFS.set(pref_name, PrefValue::Number(v)),
-            Err(_) => PREFS.set(pref_name, PrefValue::String(value.to_string())),
+    let pref_value = parse_cli_pref_value(split.get(1).cloned());
+    prefs::pref_map()
+        .set(pref_name, pref_value)
+        .expect(format!("Error setting preference: {}", pref).as_str());
+}
+
+fn parse_cli_pref_value(input: Option<&str>) -> PrefValue {
```

Comment:
Another potential problem I should point out is that the CLI args are parsed first into a `PrefValue` before updating the preference. This isn't ideal because there are some edge cases, where the incorrect variant of `PrefValue` could be parsed and then the update would fail. For example if you wanted to set a preference of type `String` with the value `"1.0"`, this would be parsed as `PrefValue::Float(1.0)` and then would panic when trying to set the `String` preference.

The _ideal_ solution here would be to somehow infer the type of the actual preference when parsing the CLI argument, but that would be a lot more complex. I would like to track that as a separate issue rather than block this PR, if that is acceptable.

Reply:
That's fine.